frontend/src/utils/helpers.js

```diff
@@ -95,3 +95,46 @@ export const getFiscalYear = () => {
 };
 
 export const formatParamStringToArray = (param) => (param ? param.split(",").filter((x) => x) : []);
+
+// This method sorts codes of the for '#.#.# - Lorem Ipsum'
+// where the number of integers is variable and the text is optional
+// TODO: In order to sort incidents this will need to be modified to deal with
+// parentheses
+export const compareCodes = (a, b) => {
+  // Returns the first match that is non-null.
+  const regexParse = (input) =>
+    input.match(/(?<first>[0-9]+)\.(?<second>[0-9]+)\.(?<third>[0-9]+)/) ||
+    input.match(/(?<first>[0-9]+)\.(?<second>[0-9]+)/) ||
+    input.match(/(?<first>[0-9]+)/);
+  const aCodes = regexParse(a);
+  const bCodes = regexParse(b);
+  if (!aCodes) return -1;
+  if (!bCodes) return 1;
+  let aInt;
+  let bInt;
+  const k = Math.min(aCodes.length, bCodes.length);
+  // Compares then non-null parts of two strings of potentially different lengths e.g 1.11 and 1.4.12
+  for (let i = 1; i < k; i += 1) {
+    aInt = parseInt(aCodes[i], 10);
+    bInt = parseInt(bCodes[i], 10);
+    if (aInt > bInt) return 1;
+    if (aInt < bInt) return -1;
+  }
+  // If the previous loop did not return anything and the codes are the same length then they are the same code.
+  if (aCodes.length === bCodes.length) {
+    return 0;
+  }
+  // The shorter of two codes with otherwise matching numbers is ordered first e.g 1.11 is before 1.11.12
+  return aCodes.length < bCodes.length ? -1 : 1;
+};
+
+export const codeSorter = (codeOne, codeTwo) => {
+  // if a code is null it set before one that is not in sort order.
```

Comment:
I think this is meant to say something like `if a code is null, set it before the other code`? As-written, I don't understand it.

Reply:
I english gooder